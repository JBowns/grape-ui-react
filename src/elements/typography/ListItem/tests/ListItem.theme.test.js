import React from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import { ListItem } from '..';
import 'jest-styled-components';

describe('ListItem using Theme color', () => {
  it('should return a ListItem object with custom color', () => {
    const theme = {
      colors: {
        grapeSoda: {
          base: 'hsl(325, 84.6%, 28%)',
          dark: 'hsl(305, 33.9%, 23.7%)',
          light: 'hsl(313, 67.8%, 47.5%)',
        },
      },
    };
    const element = (
      <ThemeProvider theme={theme}>
        <ListItem bg="grapeSoda.dark" color="grapeSoda.light">Lorem Ipsum</ListItem>
      </ThemeProvider>
    );
    const component = renderer.create(element);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('ListItem using Theme Font Family', () => {
  it('should return a ListItem object with a custom font family', () => {
    const theme = {
      fonts: {
        trueSpace: '"DejaVu Sans Mono", "Bitstream Vera Sans Mono", Courier',
      },
    };
    const element = (
      <ThemeProvider theme={theme}>
        <ListItem fontFamily="trueSpace">Lorem Ipsum</ListItem>
      </ThemeProvider>
    );
    const component = renderer.create(element);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
