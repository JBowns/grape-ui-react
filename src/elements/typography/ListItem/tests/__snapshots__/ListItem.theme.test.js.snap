// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem using Theme Font Family should return a ListItem object with a custom font family 1`] = `
.c0 {
  color: inherit;
  background-color: transparent;
  font-family: "DejaVu Sans Mono","Bitstream Vera Sans Mono",Courier;
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin-bottom: 0.25rem;
}

<li
  className="c0"
>
  Lorem Ipsum
</li>
`;

exports[`ListItem using Theme color should return a ListItem object with custom color 1`] = `
.c0 {
  color: hsl(313,67.8%,47.5%);
  background-color: hsl(305,33.9%,23.7%);
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin-bottom: 0.25rem;
}

<li
  className="c0"
>
  Lorem Ipsum
</li>
`;
